### ISSUE TITLE
CNTK: Caught Win32 exception 0xc0000005: Access violation.

### ISSUE THREAD

**Issue by vidyatshetty** [Microsoft/CNTK#2226]
Hi,
I am using cntk to build speaker recognition system. I used htk toolkit to extract mfcc features and create .mlf files. When i try to run i am getting the error
CNTK: Caught Win32 exception 0xc0000005: Access violation.
I am assuming i have problem with reading .mlf file because it shows zero entries in .mlf file though i have entries.

![cntk](https://user-images.githubusercontent.com/13535872/29395757-cc0b89d2-82d8-11e7-93a1-c1145ce5cbf3.PNG)

my config file looks like this,

command=TimitTrainSimple:TimitTestSimple:TimitWriteSimple
# command=TimitWriteSimple

# deviceId=-1 for CPU, >=0 for GPU devices
DeviceNumber=0
#stderr=Demo

#precision=float
precision = float
traceLevel = 0

#######################################
#  TRAINING CONFIG (Simple, Fixed LR) #
#######################################

TimitTrainSimple=[
    action=train

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models/TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    traceLevel=0

    # Notation xxx:yyy*n:zzz is equivalent to xxx, then yyy repeated n times,
    #  then zzz
    # example: 10:20*3:5 is equivalent to 10:20:20:20:5
    SimpleNetworkBuilder=[
        layerSizes=39:50*3:2
        trainingCriterion=CrossEntropyWithSoftmax
        evalCriterion=ClassificationError
        layerTypes=Sigmoid
        initValueScale=1.0
        applyMeanVarNorm=true
        uniformInit=true
        needPrior=true
        ]
    SGD=[
        epochSize=0 
        #means epochSize is the size of the training set
        #epochSize=20480
        minibatchSize=256:1024
        learningRatesPerMB=0.8:3.2*14:0.08
        momentumPerMB=0.9
        dropoutRate=0.0
        maxEpochs=25		# Full experiment
        #maxEpochs=1		# Small experiment for debugging
    ]
       
    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

     #readMethod=blockRandomize
      readMethod=rollingWindow

      miniBatchMode=partial
      verbosity=1
      randomize=auto
      frameMode=true
       
      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\scriptFile.scp
      ]
    

      labels=[
      type=category
	mlfFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\scriptFile.mlf.txt
	labelMappingFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\TimitStateList.txt
	
	labelDim=2
	labelType=category
      ]
    ]
]
 

#######################################
#  TESTING CONFIG (Simple)            #
#######################################

TimitTestSimple=[
    action=test

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models/TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    traceLevel=1

	# epochSize=0 means epochSize is the size of the training set
        epochSize=20480
        minibatchSize=256:1024
       
    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

      #readMethod=rollingWindow

      miniBatchMode=Partial
      randomize=Auto
      verbosity=1   

	# numMBsToShowResult=10
	# evalNodeNames

      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\testing.txt
      ]

      labels=[
	mlfFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\testingMlf.txt
	labelMappingFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\TimitStateList.txt
	labelDim=2
	labelType=category
      ]
    ]

]

#######################################
#  OUTPUT RESUTLS (Simple)            #
#######################################

TimitWriteSimple=[
    action=write

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models/TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    outputNodeNames=ScaledLogLikelihood
    traceLevel=1

    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\testing.scp
      ]

    ]
    writer=[
      wrecords=50000 #Number of records to allocate space for in file
      writerType=HTKMLFWriter
      ScaledLogLikelihood = [
          dim = 2
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Output\CntkTimitOutput.scp
           type=Real			# Must be uppercase Real?
	]
    ]
]

**Comment by vidyatshetty** [comment]
I found the fix. There was some minor error in my .mlf file my bad. It is working fine now. Thanks
